src/deluge/gui/ui/ui.cpp

```diff
@@ -55,23 +56,23 @@ void UI::close() {
 	closeUI(this);
 }
 
-#define UI_NAVIGATION_HISTORY_LENGTH 16
-UI* uiNavigationHierarchy[UI_NAVIGATION_HISTORY_LENGTH];
+constexpr size_t kUiNavigationHistoryLength = 16;
+static std::array<UI*, kUiNavigationHistoryLength> uiNavigationHierarchy;
 
-int8_t numUIsOpen = 0; // Will be 0 again during song load / swap
+int32_t numUIsOpen = 0; // Will be 0 again during song load / swap
 
-UI* lastUIBeforeNullifying = NULL;
-
-void getUIGreyoutRowsAndCols(uint32_t* cols, uint32_t* rows) {
-	*cols = 0;
-	*rows = 0;
+UI* lastUIBeforeNullifying = nullptr;
 
+std::pair<uint32_t, uint32_t> getUIGreyoutRowsAndCols() {
```

Comment:
semantically I feel like this should be a struct with rows and columns, I'm not super set on that though if there's a reason not to

Reply:
yeah either one is probably fine, I just feel like a struct is probably overkill here since it won't get used anywhere else.
The details of the return type should be documented in doxygen and that would take care of any confusion, ideally.
...Maybe I should just do that >.>

Reply:
Documented.